src/nccl_ofi_topo.cpp

```diff
@@ -1470,41 +1435,34 @@ static int write_nic(hwloc_obj_t node, FILE *file, int indent)
 	union hwloc_obj_attr_u *attr = node->attr;
 
 	/* Retrieve link speed and width of NIC */
-	if ((ret = get_pci_device_min_speed(node, true, &speed_idx, &width))) {
+	if ((ret = get_pci_device_speed(node, true, &speed_idx, &width))) {
 		NCCL_OFI_WARN("Failed to retrieve PCI speed and width of NIC");
 		return ret;
 	}
 
+	NCCL_OFI_TRACE(NCCL_INIT, "Starting NIC information: group size \"%d\", NIC speed \"%s\",  width \"%zu\"", group_size, pcie_gen[speed_idx], width);
+
 	/* Scale NIC speed and width up to the speed of the GPU. Since
 	 * NICs are grouped, GPU topology node is attached to the NIC
 	 * topology node. */
 	if (group_size > 1) {
-		hwloc_obj_t gpu = userdata->gpu_group_node;
-		size_t gpu_width;
-		size_t gpu_speed_idx;
-		assert(gpu);
-
-		/* Retrieve link speed and width of GPU */
-		if ((ret = get_pci_device_min_speed(gpu, false, &gpu_speed_idx, &gpu_width))) {
-			NCCL_OFI_WARN("Failed to retrieve PCI speed and width of GPU associated to NIC");
-			return ret;
-		}
-
 		/* In case we have multiple NICs in the group,
 		 * increase link speed if possible and decrease NIC
 		 * group size */
-		while (group_size > 1 && speed_idx < gpu_speed_idx) {
+		size_t num_pcie_gens = sizeof(pcie_gen) / sizeof(pcie_gen[0]);
+		while (group_size > 1 && speed_idx < num_pcie_gens) {
```

Comment:
Ugh, I just remembered part of why the GPU check code is here.  we have to support older NCCL that might not have Gen6 speeds and feeds.  On P5en, we want to present as Gen5x16, not Gen6x8.  With this code, we'd present as Gen6x8, so any NCCL version that doesn't have a Gen6 entry breaks.  We can assume that any NCCL version that supports a given GPU supports that GPU's PCIe generation.  But we should just use the GPU's reported max speeds, not the min of the gpu / uplink port min of curr/max.